File: 프로그래머스/Level2/[3차] n진수 게임.py
n = 16  # 진법
t = 16  # 구할 숫자 개수
m = 2  # 참가인원
p = 1  # 순서


def change_to_n(num, n):
    s = ''
    dic = {10: 'A', 11: 'B', 12: 'C', 13: 'D', 14: 'E', 15: 'F'}
    if num == 0:
        return '0'
    while num > 0:
        if num % n < 10:
            s = str(num % n) + s
        else:
            s = dic[num % n] + s
        num = num//n
    return s


def solution(n, t, m, p):
    answer = ''
    tmp = [change_to_n(i, n)for i in range(0, t*m)]
    # print(tmp)
    seq = []
    for el in tmp:  # 전체 숫자를 원소별로 나열
        dis = list(el)
        if len(dis) > 1:
            for n in dis:
                seq.append(n)
        else:
            seq.append(el)
    # print(seq)

    i = p-1
    while i < len(seq):
        answer += seq[i]
        i += m
        if len(answer) == t:
            break
    print(answer)
    return answer


solution(n, t, m, p)

File: Javascript/프로그래머스/Level2/[3차] n진수 게임.js
n = 16;
t = 16;
m = 2;
p = 1;

function change(num, n) {
    let s = '';
    const Map = { 10: 'A', 11: 'B', 12: 'C', 13: 'D', 14: 'E', 15: 'F' };
    if (num === 0) return '0';
    else {
        while (1) {
            if (num >= n) {
                if (num % n >= 10) {
                    s = Map[String(num % n)] + s;
                } else {
                    s = String(num % n) + s;
                }
                num = parseInt(num / n);
                // console.log(`s: ${s} num: ${num}`);
            } else {
                if (num % n >= 10) {
                    s = Map[String(num % n)] + s;
                } else {
                    s = String(num % n) + s;
                }
                break;
            }
        }
    }
    return s;
}

function solution(n, t, m, p) {
    var answer = '';
    let i = 0; // 부르는 숫자의 10진법
    let s = '';
    if (p === m) p = 0;
    while (1) {
        if (s.length > t * m) break;
        s += change(i, n);
        i++;
    }

    for (let i = 1; i <= s.length; i++) {
        if (i % m === p) {
            answer += s[i - 1];
        }
        if (answer.length === t) break;
    }
    console.log(answer);
    return answer;
}

solution(n, t, m, p);
